_assets/js/main.js

```diff
@@ -1007,59 +1032,93 @@ jQuery(document).ready(function($) {
+      var query_output =
+        num + ' ' + $container.data('message') + ' "' + query + '"';
       var $title = $('<h2 class="no-underline"></h2>');
 
-      // Show results
-      $container.empty();
-
-      // Add status
-      $title.text(query_output);
-      $container.prepend($title);
-
-      if (num != 0) {
-        var $list = $('<ol></ol>');
-
-        // Loop through, match, and add results --> this will be implemented when we integrate lunr js
-        // for (var item in result) {
-        //   var ref = result[item].ref;
-        //   var topic = store[ref].topic
-        //     ? '<p class="post-meta"><small>' + store[ref].topic + '</small></p>'
-        //     : '';
-        //   var searchitem =
-        //     '<li class="result"><a href="' +
-        //     store[ref].link +
-        //     '">' +
-        //     topic +
-        //     '<h3>' +
-        //     store[ref].title +
-        //     '</h3><p>' +
-        //     store[ref].excerpt +
-        //     '</p><p><small>' +
-        //     store[ref].link +
-        //     '</small></p></a></li>';
-
-        //   $list.append(searchitem);
-        // }
-
-        $default.hide(0);
-        $container.append($list);
-      } else {
-        $default.show(0);
-      }
+      data.then(function(loaded_data) {
+        // Show results
+        $container.empty();
+
+        // Add status
+        $title.text(query_output);
+        $container.prepend($title);
+
+        if (num != 0) {
+          var $list = $('<ol></ol>');
+
+          // Loop through, match, and add results
+          for (var item in result) {
+            var ref = result[item].ref;
+            var item = loaded_data.entries[ref];
+            var topic =
+              item.meta && item.meta.topic
+                ? '<p class="post-meta"><small>' +
+                  item.meta.topic +
+                  '</small></p>'
+                : '';
+            var excerpt =
+              $.trim(item.body)
+                .substring(0, 300)
+                .split(' ')
+                .slice(0, -1)
+                .join(' ') + '...';
+            var searchitem =
+              '<li class="result"><a href="' +
+              item.url +
+              '">' +
+              topic +
+              '<h3>' +
+              item.title +
+              '</h3><p>' +
+              excerpt +
+              '</p><p><small>' +
+              window.location.origin +
+              item.url +
+              '</small></p></a></li>';
+
+            $list.append(searchitem);
+          }
+
+          $default.hide(0);
+          $container.append($list);
+        } else {
+          $default.show(0);
+        }
+      });
     }
 
-    $local_input.on('keyup', function() {
-      lunr_search($(this).val());
+    // Build Lunr
+    data.then(function(loaded_data) {
+      idx = lunr(function() {
+        var l = this;
+        $.each(settings.search_fields, function(i, obj) {
```

Comment:
This is still building the lunr index at runtime, it should be built offline as part of the buld process. This would negate the need to publish the json from the pages API and you can just pre-build an index at build time.

Reply:
Maybe https://github.com/18F/jekyll_pages_api_search#jekyll_pages_api_search-plugin could be used?

Reply:
@muffinresearch we could use the Jekyll pages api search plugin, but we would lose having  content snippets on the results listing. We'd also lose our tag search solution. My first approach to implementing search was this plugin. I tried forking it to make modifications to allow content snippets, but it was failing to build locally: https://github.com/18F/jekyll_pages_api_search/issues/46

Reply:
If that plugin isn't suitable, this can just be a node script run after the jekyll build - and that way you can easily include your specializations there.

Reply:
This update could be a follow-up if that makes things easier?

Reply:
Thanks, yes that would be great.